### PR TITLE
Add `/user` filter to the websocket stream

### PR DESCRIPTION
**Pull request by esanzgar**
This PR adds a new filter capability to the websocket stream. It allows to filter by annotations from specific users.
    
I reorder the test to match the filters as they appear in the code: id -> group -> user -> uri -> references

### CODE REVIEW

File: tests/h/streamer/filter_test.py [hunk 1]
```diff
@@ -13,6 +13,55 @@ class FakeSocket:
 
 
 class TestFilterHandler:
+
```
Comment (esanzgar) [inline tests/h/streamer/filter_test.py:16]: Change the order of test to match the code: /id, /group, /user, /uri, /references

File: tests/h/streamer/filter_test.py [hunk 2]
```diff
@@ -218,11 +208,19 @@ def get_randomized_filter(self):  # pragma: no cover
...
+                        "333QnPxg" + str(random()),
+                        "333QnPxg" + str(random()),
+                    ],
+                },
+                {
+                    "field": "/user",
+                    "operator": "equals",
+                    "value": [
+                        "acct:@" + str(random()),
+                        "acct:@" + str(random()),
```
Comment (robertknight) [inline tests/h/streamer/filter_test.py:223]: I suggest formatting this as a valid userid (`acct:<non-empty username>@<authority>`).

**Review by robertknight — APPROVED** [review]
The changes related to the user filter look good. It would be easier to review and for future devs to reference if the changes that reordered the tests were split out into a separate commit. I made a note that the current username matching implementation only accepts an exact `acct:<username>@<authority>` string whereas the search API accepts just a username. I suggest landing this implementation first, but it would be good to follow up with a change that makes `user` handling consistent with the search API.

**Comment by esanzgar** [comment]
This change was to support a filtering feature in the notebook frontend project. We realised that because there are other filtering capabilities that will be added in the future, we can't keep modifying the websocket filtering for each one of them. It is probably better to filter in the client side.

**Pull request closed without merging**